src/SourceBuild/content/repo-projects/package-source-build.proj

```diff
@@ -9,61 +9,58 @@
           AfterTargets="RepoBuild"
           DependsOnTargets="CreateBuildInputProps;DetermineSourceBuiltSdkVersion">
-    <PropertyGroup>
-      <SourceBuildTarballStagingDir>$([MSBuild]::NormalizeDirectory('$(BaseIntermediateOutputPath)', '$(SourceBuiltArtifactsTarballName)'))</SourceBuildTarballStagingDir>
-      <SourceBuildReferencePackagesDestination>$([MSBuild]::NormalizeDirectory('$(SourceBuildTarballStagingDir)', 'SourceBuildReferencePackages'))</SourceBuildReferencePackagesDestination>
-    </PropertyGroup>
-
-    <!-- Copy PVP to staging dir in order to package them together. -->
-    <Copy SourceFiles="$(CurrentSourceBuiltPackageVersionPropsPath)"
-          DestinationFiles="$(SourceBuildTarballStagingDir)PackageVersions.props" />
-
-    <ItemGroup>
-      <SourceBuildReferencePackagesNupkgFiles Include="$(ReferencePackagesDir)**/*.nupkg" />
-    </ItemGroup>
-
-    <!-- Copy reference packages from ReferencePackagesDir to staging dir. -->
-    <Copy
-      SourceFiles="@(SourceBuildReferencePackagesNupkgFiles)"
-      DestinationFiles="@(SourceBuildReferencePackagesNupkgFiles -> '$(SourceBuildReferencePackagesDestination)%(Filename)%(Extension)')"
-      Condition="'@(SourceBuildReferencePackagesNupkgFiles)' != ''" />
-
-    <ItemGroup>
-      <ArtifactsPackage Include="$(ArtifactsPackagesDir)**\*.nupkg" />
-    </ItemGroup>
-
-    <!-- Copy packages to staging dir. -->
-    <Copy SourceFiles="@(ArtifactsPackage)"
-          DestinationFolder="$(SourceBuildTarballStagingDir)"
-          Condition="'@(ArtifactsPackage)' != ''" />
-
     <PropertyGroup>
       <SourceBuiltTarballName>$(ArtifactsAssetsDir)$(SourceBuiltArtifactsTarballName).$(SourceBuiltSdkVersion).$(TargetRid)$(ArchiveExtension)</SourceBuiltTarballName>
       <SourceBuiltVersionFileName>.version</SourceBuiltVersionFileName>
     </PropertyGroup>
 
-    <!-- Content of the .version file to include in the tarball. Write to staging dir. -->
+    <!-- Content of the .version file to include in the tarball -->
     <ItemGroup>
       <VersionFileContent Include="$(RepositoryCommit);$(SourceBuiltSdkVersion)" />
     </ItemGroup>
 
     <WriteLinesToFile
-      File="$(SourceBuildTarballStagingDir)$(SourceBuiltVersionFileName)"
+      File="$(BaseIntermediateOutputPath)$(SourceBuiltVersionFileName)"
       Lines="@(VersionFileContent)"
       Overwrite="true" />
 
-    <MakeDir Directories="$([System.IO.Path]::GetDirectoryName('$(SourceBuiltTarballName)'))" />
-    <Exec Command="tar --numeric-owner -czf $(SourceBuiltTarballName) $(SourceBuiltVersionFileName) *.nupkg *.props SourceBuildReferencePackages/"
-          WorkingDirectory="$(SourceBuildTarballStagingDir)" />
+    <ItemGroup>
+      <_AllRepoDirs Include="$([System.IO.Directory]::GetDirectories($(ArtifactsShippingPackagesDir)))" />
+      <_AllRepoDirs Include="$([System.IO.Directory]::GetDirectories($(ArtifactsNonShippingPackagesDir)))" />
+      <_AllRepoFiles Include="%(_AllRepoDirs.Identity)/**" />
```

Comment:
Why do you need this? The Copy task with the DestinationFolder metadata already does a flat copy.

Reply:
I'm not sure I'm following. The `Copy` task requires a list of files. That's what this does. I can't just provide it directory paths.

Reply:
I can simply this to this:

```xml
<_AllRepoFiles Include="$(ArtifactsShippingPackagesDir)/**" />
<_AllRepoFiles Include="$(ArtifactsNonShippingPackagesDir)/**" />
```

Is that what you meant?